tests/testsuite/publish.rs

```diff
@@ -1706,3 +1734,73 @@ Caused by:
 
     t.join().unwrap();
 }
+
+#[cargo_test]
+fn in_workspace() {
+    registry::init();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [workspace]
+                members = ["foo", "bar"]
+            "#,
+        )
+        .file(
+            "foo/Cargo.toml",
+            r#"
+                [project]
+                name = "foo"
+                version = "0.0.1"
+                authors = []
+                license = "MIT"
+                description = "foo"
+            "#,
+        )
+        .file("foo/src/main.rs", "fn main() {}")
+        .file(
+            "bar/Cargo.toml",
+            r#"
+                [project]
+                name = "bar"
+                version = "0.0.1"
+                authors = []
+                license = "MIT"
+                description = "bar"
+                workspace = ".."
+            "#,
+        )
+        .file("bar/src/main.rs", "fn main() {}")
+        .build();
+
+    p.cargo("publish --no-verify --token sekrit -p foo")
+        .with_stderr(&format!(
+            "\
+[UPDATING] `{reg}` index
```

Comment:
The exacting text of the `Updating` line isn't important to check here, so I think something like `[UPDATING] [..]` should be fine (without the format!).